cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -70,18 +73,26 @@ void BM_contiguous_split(benchmark::State& state)
   int64_t num_rows        = total_desired_bytes / (num_cols * el_size);
 
   // generate input table
-  srand(31337);
-  auto valids = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return true; });
-  std::vector<cudf::test::fixed_width_column_wrapper<int>> src_cols(num_cols);
+  auto valids = cudf::detail::make_counting_transform_iterator(
+    0, [] __device__(cudf::size_type i) { return true; });
+  std::vector<std::unique_ptr<cudf::column>> src_cols;
+  src_cols.reserve(num_cols);
   for (int idx = 0; idx < num_cols; idx++) {
```

Comment:
I wonder if it would make more sense to write a standalone (or lambda) function like
```c++
template<typename T>
auto make_random_column(cudf::size_type num_rows, int seed, bool include_validity) {
    // Compute random values
    return cudf::test::fixed_width_column_wrapper<T>(...);
}
```
to separate the column generation logic from the wrapping `for` loop that is assigning those results to the elements of `src_cols`. Then we could use an algorithm instead of the raw loop to fill the vector of generated columns.

Reply:
It seems like the logic in that function could be re-used for generating the string columns below, too, if it accepted a parameter to control the maximum value of the generated random distribution?